src/DynamoCore/Models/RecordableCommands.cs

```diff
@@ -689,6 +689,110 @@ protected override void SerializeCore(XmlElement element)
             #endregion
         }
 
+        [DataContract]
+        public class AutoCreateNodeCommand : ModelBasedRecordableCommand
+        {
+            #region Public Class Methods
+
+            private void SetProperties(int outputPortIndex, int inputPortIndex, double x, double y, 
+                bool defaultPosition, bool transformCoordinates)
+            {
+                OutputPortIndex = outputPortIndex;
+                InputPortIndex = inputPortIndex;
+                X = x;
+                Y = y;
+                DefaultPosition = defaultPosition;
+                TransformCoordinates = transformCoordinates;
+            }
+
+            public AutoCreateNodeCommand(
+                NodeModel inputNode, NodeModel givenNode, int outputPortIndex, int inputPortIndex, 
+                double x, double y, bool defaultPosition, bool transformCoordinates)
+                : base(inputNode != null && givenNode != null ? new[] { inputNode.GUID, givenNode.GUID } : new[] { Guid.Empty })
```

Comment:
I'm imagining the signature of this constructor as followed:

```csharp
public CreateAndConnectNodeCommand(
    string existingNodeId,      // Guid of an existing node to connect from/to
    string newNodeId,           // Guid to assign to the newly created node
    string newNodeName,         // The name of node to create (as in CreateNodeCommand)
    int outPortIndex,           // Output port index
    int inPortIndex,            // Input port index
    double x,                   // New node location
    double y,                   // New node location
    bool createAsDownstreamNode // The new node is a downstream node (having an input port)
);
```

Few points:

1. We don't pass in just `NodeModel`, because that cannot be deserialized (the commands that do it have different purposes), these parameters need to be serializable (i.e. primitive types like `string`).
2. We foresee this command be used for your purposes (and also an upcoming user story), so we will just make `defaultPosition` and `transformCoordinates` as `false` always, and remove them from the signature.